File: src/Sample/Plumix.Sample/Demos/Cupertino/CupertinoActivityIndicatorDemoPage.cs
using Avalonia.Media;
using Plumix.Cupertino;
using Plumix.Rendering;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source (reference): dart_sample/lib/demos/cupertino/cupertino_activity_indicator_demo_page.dart
// (exact sample parity)

public sealed class CupertinoActivityIndicatorDemoPage : StatefulWidget
{
    public override State CreateState() => new CupertinoActivityIndicatorDemoPageState();
}

internal sealed class CupertinoActivityIndicatorDemoPageState : State
{
    private double _progress = 0.6;

    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 12.0,
            children:
            [
                new Text("Cupertino activity indicators", fontSize: 20.0, color: Colors.Black),
                new Text(
                    "Spinning ticks, partially revealed ticks and the linear progress bar.",
                    fontSize: 14.0,
                    color: Color.Parse("#8A000000")),
                new Wrap(
                    spacing: 24.0,
                    runSpacing: 12.0,
                    crossAxisAlignment: WrapCrossAlignment.Center,
                    children:
                    [
                        BuildLabeled("Default", new CupertinoActivityIndicator()),
                        BuildLabeled("radius 20", new CupertinoActivityIndicator(radius: 20.0)),
                        BuildLabeled(
                            "Tinted",
                            new CupertinoActivityIndicator(
                                color: CupertinoColors.ActiveOrange.Value,
                                radius: 20.0)),
                        BuildLabeled(
                            "Paused",
                            new CupertinoActivityIndicator(animating: false, radius: 20.0)),
                        BuildLabeled(
                            "Partial",
                            CupertinoActivityIndicator.PartiallyRevealed(
                                progress: _progress,
                                radius: 20.0)),
                    ]),
                new Text("Progress for the partial spinner and the bars below:", fontSize: 14.0, color: Colors.Black),
                new CupertinoSlider(
                    value: _progress,
                    onChanged: value => SetState(() => _progress = value)),
                new CupertinoLinearActivityIndicator(progress: _progress),
                new CupertinoLinearActivityIndicator(
                    progress: _progress,
                    height: 10.0,
                    color: CupertinoColors.ActiveGreen.Value),
            ]);
    }

    private static Widget BuildLabeled(string label, Widget indicator)
    {
        return new Column(
            mainAxisSize: MainAxisSize.Min,
            spacing: 6.0,
            children:
            [
                new SizedBox(
                    width: 44.0,
                    height: 44.0,
                    child: new Center(child: indicator)),
                new Text(label, fontSize: 12.0, color: Colors.Black),
            ]);
    }
}

File: dart_sample/lib/demos/cupertino/cupertino_activity_indicator_demo_page.dart
import 'package:cupertino_ui/cupertino_ui.dart';
import 'package:material_ui/material_ui.dart';

class CupertinoActivityIndicatorDemoPage extends StatefulWidget {
  const CupertinoActivityIndicatorDemoPage({super.key});

  @override
  State<CupertinoActivityIndicatorDemoPage> createState() =>
      _CupertinoActivityIndicatorDemoPageState();
}

class _CupertinoActivityIndicatorDemoPageState extends State<CupertinoActivityIndicatorDemoPage> {
  double _progress = 0.6;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 12,
      children: <Widget>[
        const Text(
          'Cupertino activity indicators',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Spinning ticks, partially revealed ticks and the linear progress '
          'bar.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Wrap(
          spacing: 24,
          runSpacing: 12,
          crossAxisAlignment: WrapCrossAlignment.center,
          children: <Widget>[
            _buildLabeled('Default', const CupertinoActivityIndicator()),
            _buildLabeled(
              'radius 20',
              const CupertinoActivityIndicator(radius: 20),
            ),
            _buildLabeled(
              'Tinted',
              const CupertinoActivityIndicator(
                color: CupertinoColors.activeOrange,
                radius: 20,
              ),
            ),
            _buildLabeled(
              'Paused',
              const CupertinoActivityIndicator(animating: false, radius: 20),
            ),
            _buildLabeled(
              'Partial',
              CupertinoActivityIndicator.partiallyRevealed(
                progress: _progress,
                radius: 20,
              ),
            ),
          ],
        ),
        const Text(
          'Progress for the partial spinner and the bars below:',
          style: TextStyle(fontSize: 14, color: Colors.black),
        ),
        CupertinoSlider(
          value: _progress,
          onChanged: (double value) => setState(() => _progress = value),
        ),
        CupertinoLinearActivityIndicator(progress: _progress),
        CupertinoLinearActivityIndicator(
          progress: _progress,
          height: 10,
          color: CupertinoColors.activeGreen,
        ),
      ],
    );
  }

  Widget _buildLabeled(String label, Widget indicator) {
    return Column(
      mainAxisSize: MainAxisSize.min,
      spacing: 6,
      children: <Widget>[
        SizedBox(width: 44, height: 44, child: Center(child: indicator)),
        Text(label, style: const TextStyle(fontSize: 12, color: Colors.black)),
      ],
    );
  }
}
